apps/chat/agent/tools.py

```diff
@@ -125,21 +123,28 @@ def _move_datetime_to_new_weekday_and_time(date: datetime, new_weekday: int, new
     return date.replace(hour=new_hour, minute=new_minute, second=0) + timedelta(days=day_diff)
 
 
-def create_periodic_task(experiment_session: ExperimentSession, message: str, **kwargs):
-    task_kwargs = json.dumps(
-        {
-            "chat_ids": [experiment_session.participant.identifier],
-            "message": message,
-            "is_bot_instruction": False,
-            "experiment_public_id": str(experiment_session.experiment.public_id),
-        }
-    )
-    PeriodicTask.objects.create(
-        name=f"reminder-{experiment_session.id}-{uuid.uuid4()}",
-        task=BOT_MESSAGE_FOR_USER_TASK,
-        kwargs=task_kwargs,
-        **kwargs,
-    )
+def create_schedule_message(experiment_session: ExperimentSession, message: str, **kwargs):
```

Comment:
Looks like `cleaned_data["name"]` will fail the form validation, since it would be missing? i don't think the name here is too important, so we could probably generate it. Also would be a good idea to add a test for this function.

Reply:
[7ec855d](https://github.com/dimagi/open-chat-studio/pull/527/commits/7ec855d043b90502bb536237c2e585b95a38fdd7) modified so it gernerateds a name with a unique uuid